File: python/string_utilities.py
"""
String generation and manipulation utility functions for test automation
"""

import random
import string
import uuid


def generate_random_string(length: int = 10) -> str:
    """
    Generate a random string of specified length
    
    Args:
        length: Length of the random string
        
    Returns:
        Random alphanumeric string
    """
    characters = string.ascii_letters + string.digits
    return ''.join(random.choice(characters) for _ in range(length))


def generate_random_integer(length: int = 10) -> str:
    """
    Generate a random integer of specified length
    
    Args:
        length: Length of the random integer string
        
    Returns:
        Random integer string
    """
    return ''.join(random.choice(string.digits) for _ in range(length))


def generate_uuid() -> str:
    """
    Generate a unique identifier (UUID)
    
    Returns:
        UUID string
    """
    return str(uuid.uuid4())

File: dotNet/StringUtilities.cs
using System;
using System.Linq;

namespace TestAutomation.Utils
{
    /// <summary>
    /// String generation and manipulation utility functions for test automation
    /// </summary>
    public static class StringUtilities
    {
        /// <summary>
        /// Generate a random string of specified length
        /// </summary>
        /// <param name="length">Length of the random string</param>
        /// <returns>Random alphanumeric string</returns>
        public static string GenerateRandomString(int length)
        {
            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
            var random = new Random();
            return new string(Enumerable.Repeat(chars, length)
                .Select(s => s[random.Next(s.Length)]).ToArray());
        }

        /// <summary>
        /// Generate a random integer of specified length
        /// </summary>
        /// <param name="length">Length of the random integer string</param>
        /// <returns>Random integer string</returns>
        public static string GenerateRandomInteger(int length)
        {
            const string digits = "0123456789";
            var random = new Random();
            return new string(Enumerable.Repeat(digits, length)
                .Select(s => s[random.Next(s.Length)]).ToArray());
        }

        /// <summary>
        /// Generate a unique identifier (GUID)
        /// </summary>
        /// <returns>GUID string</returns>
        public static string GenerateGuid()
        {
            return Guid.NewGuid().ToString();
        }
    }
}
